src/models/v2/message-trait.ts

```diff
@@ -18,12 +17,17 @@ import type { SchemaInterface } from '../schema';
 import type { TagsInterface } from '../tags';
 
 import type { v2 } from '../../spec-types';
+import { getDefaultSchemaFormat } from 'schema-parser';
 
 export class MessageTrait<J extends v2.MessageTraitObject = v2.MessageTraitObject> extends BaseModel<J, { id: string }> implements MessageTraitInterface {
   id(): string {
     return this.messageId() || this._meta.id || this.json(xParserMessageName) as string;
   }
 
+  hasSchemaFormat(): boolean {
+    return true;
+  }
+
   schemaFormat(): string {
```

Comment:
```suggestion
  schemaFormat(): string | undefined {
```

Shouldn't this method return `string | undefined` to fulfill the interface?